### PR TITLE
Explain interceptor icon on mouse over

### PR DESCRIPTION
**Pull request by KillariDev**
fixes: https://github.com/DarkFlorist/TheInterceptor/issues/31

image
image

### CODE REVIEW

File: extension/app/ts/components/subcomponents/CopyToClipboard.tsx [hunk 1]
```diff
@@ -21,3 +21,21 @@ export function CopyToClipboard(props: Props): preact.VNode {
 		</div>
 	)
 }
+
+interface ToolTipParams {
+	children: preact.VNode
+	content: string
+}
+
+export function ToolTip(props: ToolTipParams): preact.VNode {
```
Comment (MicahZoltu) [inline extension/app/ts/components/subcomponents/CopyToClipboard.tsx:30]: The inferred type will likely be better than an explicit type here.
```suggestion
export function ToolTip(props: ToolTipParams) {
```

File: extension/app/ts/components/subcomponents/CopyToClipboard.tsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+interface CopyToClipboardParams {
 	children: preact.VNode
 	content: string
 	contentDisplayOverride?: string
-	copyMessage: string | undefined
+	copyMessage?: string
 }
 
-export function CopyToClipboard(props: Props): preact.VNode {
+export function CopyToClipboard(props: CopyToClipboardParams): preact.VNode {
```
Comment (MicahZoltu) [inline extension/app/ts/components/subcomponents/CopyToClipboard.tsx:10]: The inferred type will likely be better than an explicit type here.
```suggestion
export function CopyToClipboard(props: CopyToClipboardParams) {
```

Generally speaking, it is better to use inferred types over explicit types unless inference isn't doing the right thing for some reason.

File: extension/app/ts/components/subcomponents/CopyToClipboard.tsx [hunk 3]
```diff
@@ -21,3 +21,21 @@ export function CopyToClipboard(props: Props): preact.VNode {
 		</div>
 	)
 }
+
+interface ToolTipParams {
+	children: preact.VNode
```
Comment (MicahZoltu) [inline extension/app/ts/components/subcomponents/CopyToClipboard.tsx:26]: Should this be an array?  Also, I think you want `ComponentChildren` here?
https://preactjs.com/guide/v10/typescript#function-components

**Review by MicahZoltu — APPROVED** [review]

File: extension/app/ts/components/subcomponents/CopyToClipboard.tsx (hunk 3 above)
Comment (KillariDev) [reply extension/app/ts/components/subcomponents/CopyToClipboard.tsx:26]: yeah, seems like that works too. I replaced it in other places too

**Pull request merged**